src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,429 @@
+goog.provide('ol.interaction.ExtentEvent');
+goog.provide('ol.interaction.Extent');
+
+goog.require('ol');
+goog.require('ol.Feature');
+goog.require('ol.MapBrowserEvent.EventType');
+goog.require('ol.MapBrowserPointerEvent');
+goog.require('ol.coordinate');
+goog.require('ol.events');
+goog.require('ol.events.Event');
+goog.require('ol.geom.GeometryLayout');
+goog.require('ol.geom.GeometryType');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.Interaction');
+goog.require('ol.interaction.Pointer');
+goog.require('ol.layer.Vector');
+goog.require('ol.proj');
+goog.require('ol.source.Vector');
+
+/**
+ * @classdesc
+ * Events emitted by {@link ol.interaction.ExtentInteraction} instances are instances of
+ * this type.
+ *
+ * @constructor
+ * @extends {ol.events.Event}
+ */
+ol.interaction.ExtentEvent = function() {
+  goog.base(this, 'bbox');
```

Comment:
For convenience, it would make sense to set the extent on the event object.

And for consistency with other events, it would make sense to define `'bbox'` as ExtentEventType.